src/profile-logic/import/instruments/index.js

```diff
@@ -0,0 +1,910 @@
+
+          if (byteOrder !== 1) {
+            digit = ((digit & 0xff00) >> 8) | ((digit & 0x00ff) << 8);
+          }
+
+          decimal += digit * Math.pow(65536, i);
+        }
+
+        decimal *= Math.pow(10, exponent);
+        return negative ? -decimal : decimal;
+      }
+
+      // Replace NSData with a Uint8Array
+      case 'NSData':
+      case 'NSMutableData':
+        return value['NS.bytes'] || value['NS.data'];
+
+      // Replace NSString with a string
+      case 'NSString':
+      case 'NSMutableString':
+        return value['NS.bytes'] ? decodeUTF8(value['NS.bytes']) : '';
+
+      // Replace NSArray with an Array
+      case 'NSArray':
+      case 'NSMutableArray': {
+        if ('NS.objects' in value) {
+          return value['NS.objects'];
+        }
+        const array: any[] = [];
+        while (true) {
+          const object = 'NS.object.' + array.length;
+          if (!(object in value)) {
+            break;
+          }
+          array.push(value[object]);
+        }
+        return array;
+      }
+      case '_NSKeyedCoderOldStyleArray': {
+        const count = value['NS.count'];
+
+        // const size = value['NS.size']
+        // Types are encoded as single printable characters.
+        // See: https://github.com/apple/swift-corelibs-foundation/blob/76995e8d3d8c10f3f3ec344dace43426ab941d0e/Foundation/NSObjCRuntime.swift#L19
+        // const type = String.fromCharCode(value['NS.type'])
+
+        const array: any[] = [];
+        for (let i = 0; i < count; i++) {
+          const element = value['$' + i];
+          array.push(element);
+        }
+        return array;
+      }
+
+      case 'NSDictionary':
+      case 'NSMutableDictionary': {
+        const map = new Map();
+        if ('NS.keys' in value && 'NS.objects' in value) {
+          for (let i = 0; i < value['NS.keys'].length; i++) {
+            map.set(value['NS.keys'][i], value['NS.objects'][i]);
+          }
+        } else {
+          while (true) {
+            const key = 'NS.key.' + map.size;
+            const object = 'NS.object.' + map.size;
+            if (!(key in value) || !(object in value)) {
+              break;
+            }
+            map.set(value[key], value[object]);
+          }
+        }
+        return map;
+      }
+      default: {
+        const converted = interpretClass(name, value);
+        if (converted !== value) {
+          return converted;
+        }
+      }
+    }
+  }
+  return value;
+}
+
+function isDictionary(value: any): boolean {
+  return (
+    value !== null &&
+    typeof value === 'object' &&
+    Object.getPrototypeOf(value) === null
+  );
+}
+
+function isArray(value: any): boolean {
+  return value instanceof Array;
+}
+
+// This function populates data fields with given interpretClass(decided by various datatypes
+// from readInstrumentsArchive function)
+function expandKeyedArchive(
+  root: any,
+  interpretClass: ($classname: string, obj: any) => any = x => x
```

Comment:
The default value for `interpretClass` seems wrong, like above.
and like above, maybe we don't need one because we always call it with one?

Reply:
@julienw, Made the changes.